src/imageops/colorops.rs

```diff
@@ -106,6 +107,82 @@ pub fn brighten<I, P, S>(image: &I, value: i32)
 
     out
 }
+fn clamp_u8(num:f64) -> u8 {
+    if num < 0.0 {
+        return u8::min_value();
+    } else if num > 255.0 {
+        return u8::max_value();
+    } else {
+        return num as u8;
+    }
+}
+
+/// Hue rotate the supplied image.
+/// ```value``` is the degrees to rotate each pixel by.
+/// 0 and 360 do nothing, the rest rotates by the given degree value.
+/// just like the css webkit filter hue-rotate(180)
+pub fn huerotate<I, P, S>(image: &I, value: i32)
+    -> ImageBuffer<P, Vec<S>>
+    where I: GenericImage<Pixel=P>,
+          P: Pixel<Subpixel=S> + 'static,
+          S: Primitive + 'static {
+
+    let (width, height) = image.dimensions();
+    let mut out = ImageBuffer::new(width, height);
+
+    let angle: i32 = NumCast::from(value).unwrap();
+
+    let cosv = (angle as f64 * PI / 180.0).cos();
+    let sinv = (angle as f64 * PI / 180.0).sin();
+    let mut matrix: [f64; 9] = [
+        1.0, 0.0, 0.0,   // Reds
+        0.0, 1.0, 0.0,   // Greens
+        0.0, 0.0, 1.0    // Blues
+    ];
+    matrix[0] = 0.213 + cosv * 0.787 - sinv * 0.213;
+    matrix[1] = 0.715 - cosv * 0.715 - sinv * 0.715;
+    matrix[2] = 0.072 - cosv * 0.072 + sinv * 0.928;
+
+    matrix[3] = 0.213 - cosv * 0.213 + sinv * 0.143;
+    matrix[4] = 0.715 + cosv * 0.285 + sinv * 0.140;
+    matrix[5] = 0.072 - cosv * 0.072 - sinv * 0.283;
+
+    matrix[6] = 0.213 - cosv * 0.213 - sinv * 0.787;
+    matrix[7] = 0.715 - cosv * 0.715 + sinv * 0.715;
+    matrix[8] = 0.072 + cosv * 0.928 + sinv * 0.072;
+
```

Comment:
You’re initializing the matrix as _I_3_ but you aren’t using it afterwards. You can just initialize it directly, then it doesn’t have to be `mut`. Or at least initialize it as `[0; 9]` otherwise that is confusing.